### PR TITLE
Universe part

### PR DESCRIPTION
**Pull request by uldrim**

### CODE REVIEW

File: CoinGeckoMarketCapUniverseSelectionAlgorithm.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            {
+                foreach (var datum in data)
+                {
+                    Log($"{datum.Symbol},{datum.Marketcap}");
+                }
+
+                // define our selection criteria
+                return from d in data
+                       where d.Marketcap > 200000
+                       select d.Symbol;
```
Comment (AlexCatarino) [inline CoinGeckoMarketCapUniverseSelectionAlgorithm.cs]: We cannot return the Symbol, but a combination of the Base Currency (Coin) and the Quote Currency (e.g. USD) to create a `Symbol`.

File: CoinGeckoMarketCapUniverse.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        {
+            return true;
+        }
+
+        /// <summary>
+        /// Converts the instance to string
+        /// </summary>
+        public override string ToString()
+        {
+            return $"{Symbol} - Marketcap : {Marketcap}";
```
Comment (AlexCatarino) [inline CoinGeckoMarketCapUniverse.cs]:
```csharp
return $"{Coin}.{Symbol} - Marketcap : {Marketcap}";
```

File: CoinGeckoMarketCapUniverse.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+using System.Globalization;
+
+namespace QuantConnect.DataSource
+{
+    /// <summary>
+    /// Example custom data type
+    /// </summary>
+    [ProtoContract(SkipConstructor = true)]
+    public class CoinGeckoMarketCapUniverse : BaseData
+    {
```
Comment (AlexCatarino) [inline CoinGeckoMarketCapUniverse.cs:32]:
```csharp
        /// <summary>
        /// Coin name
        /// </summary>
        public string Coin { get; set; }
```

File: CoinGeckoMarketCapUniverse.cs [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        /// <summary>
+        /// Marketcap of the coin
+        /// </summary>
+        public decimal Marketcap { get; set; }
+
+        /// <summary>
+        /// Time passed between the date of the data and the time the data became available to us
+        /// </summary>
+        public TimeSpan Period { get; set; } = TimeSpan.FromDays(1);
```
Comment (AlexCatarino) [inline CoinGeckoMarketCapUniverse.cs]:
```csharp
private TimeSpan _period = TimeSpan.FromDays(1);
```

**Review by AlexCatarino — CHANGES_REQUESTED** [review]
Thank you, @kuruma99.
The main issue is not adding the coin name. Also, please add sample data, and test the algorithm.

**Comment by uldrim** [comment]
I have backtested the universe algorithm and the output is correct. Attaching the result of the demo algo along with the raw data retrieved from the CoinGecko API(in the output directory).
The dates used in the demo algo are Feb 14'2022 - Feb 18' 2022
output
btc_universe
eth_universe

**Comment by uldrim** [comment]
The final output of the demo algorithm.
output2
The output is consistent with the data files in https://github.com/kuruma99/Lean.DataSource.CoinGeckoMarketCap/tree/master/output/alternative/coingecko/marketcap/universe

**Pull request merged**